test/test_status_event_repository.rb

```
@@ -100,56 +100,86 @@ def test_get_latest_event_for_bag_when_nil
-    mixin_bag_repo.create(identifier: bag_identifier_three, group_part: 1, repository_package_identifier: three_package_identifier)
-    mixin_repo.create(status: BagStatus::COPYING, bag_identifier: bag_identifier_one, timestamp: start_time - 30)
-    mixin_repo.create(status: BagStatus::COPIED, bag_identifier: bag_identifier_one, timestamp: start_time - 60)
-    mixin_repo.create(status: BagStatus::COPYING, bag_identifier: bag_identifier_one, timestamp: start_time)
-    mixin_repo.create(status: BagStatus::COPIED, bag_identifier: bag_identifier_one, timestamp: start_time + 30)
-    mixin_repo.create(status: BagStatus::COPYING, bag_identifier: bag_identifier_one, timestamp: start_time + 60)
-    mixin_repo.create(status: BagStatus::DEPOSITED, bag_identifier: bag_identifier_one, timestamp: start_time + 90)
-    mixin_repo.create(status: BagStatus::COPYING, bag_identifier: bag_identifier_two, timestamp: start_time + 100)
-    mixin_repo.create(status: BagStatus::COPIED, bag_identifier: bag_identifier_two, timestamp: start_time + 120)
-    mixin_repo.create(status: BagStatus::COPYING, bag_identifier: bag_identifier_three, timestamp: start_time + 140)
-    mixin_repo.create(status: BagStatus::FAILED, bag_identifier: bag_identifier_three, timestamp: start_time + 160)
-    mixin_repo.create(status: BagStatus::COPYING, bag_identifier: bag_identifier_two, timestamp: start_time + 180)
-    mixin_repo.create(status: BagStatus::FAILED, bag_identifier: bag_identifier_two, timestamp: start_time + 200)
+  def test_get_latest_event_for_bags_when_no_events
+    start = Time.utc(2024, 5, 4, 12, 0, 0, 0)
+    bag_events = mixin_repo.get_latest_event_for_bags(start_time: start)
+    assert_equal 0, bag_events.length
+  end
 
+  def test_get_latest_event_for_bags
+    bag_id_one, bag_id_two, bag_id_three, bag_id_four = create_latest_event_for_bags_test_data
```

bag_id_one can be removed since its not used

---

I'm not sure how to do this when you're destructuring. Let me check.

---

Fixed. See https://github.com/mlibrary/bag-courier/pull/53/commits/56c048da06d398d13aba9e075e783fc68de3eaee

---

if i remove bag_id_one its failing the test but i get that variable unused error. can we check the assert for bag_id_one or we can ignore this error?

---

I'm pretty sure I fixed it in my last commit. GitHub says tests are passing. Are you using the latest code? Or are you changing the wrong test? Cause the second one uses all four variables.

---

works with the latest